ast/desugar/Desugar.cc

```diff
@@ -617,6 +617,17 @@ class DuplicateHashKeyCheck {
     }
 };
 
+core::NameRef freshNameFromRef(DesugarContext dctx, const ExpressionPtr &ref) {
+    if (auto *nm = cast_tree<UnresolvedIdent>(ref)) {
+        return dctx.ctx.state.freshNameUnique(core::UniqueNameKind::Desugar, nm->name, ++dctx.uniqueCounter);
+    }
+    if (auto *nm = cast_tree<ast::Local>(ref)) {
+        return dctx.ctx.state.freshNameUnique(core::UniqueNameKind::Desugar, nm->localVariable._name,
+                                              ++dctx.uniqueCounter);
+    }
+    Exception::notImplemented();
```

Comment:
I generally don’t like this because half the time when it gets tripped there’s no backtraces or line numbers and it’s annoying to get one CS if there had just been a mention of the function name in the error message. So I always raise my own error with a unique looking message